plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -344,26 +344,40 @@ public Stream<AddFileEntry> getActiveFiles(
             MetadataEntry metadataEntry,
             ProtocolEntry protocolEntry,
             TupleDomain<DeltaLakeColumnHandle> partitionConstraint,
+            TupleDomain<DeltaLakeColumnHandle> nonPartitionConstraint,
             Set<DeltaLakeColumnHandle> projectedColumns)
     {
         Set<String> baseColumnNames = projectedColumns.stream()
                 .filter(DeltaLakeColumnHandle::isBaseColumn) // Only base column stats are supported
                 .map(DeltaLakeColumnHandle::columnName)
                 .collect(toImmutableSet());
-        return getActiveFiles(session, tableSnapshot, metadataEntry, protocolEntry, partitionConstraint, baseColumnNames::contains);
+        return getActiveFiles(session, tableSnapshot, metadataEntry, protocolEntry, unionDomains(partitionConstraint, nonPartitionConstraint), baseColumnNames::contains);
+    }
+
+    public static TupleDomain<DeltaLakeColumnHandle> unionDomains(TupleDomain<DeltaLakeColumnHandle> enforcedPartitionConstraint, TupleDomain<DeltaLakeColumnHandle> nonPartitionConstraint)
+    {
+        if (enforcedPartitionConstraint.isAll()) {
+            return nonPartitionConstraint;
+        }
+
+        if (nonPartitionConstraint.isAll()) {
+            return enforcedPartitionConstraint;
+        }
+
+        return TupleDomain.columnWiseUnion(enforcedPartitionConstraint, nonPartitionConstraint);
```

Comment:
I think this should rather be intersection of TupleDomain

Reply:
Amazing, thanks for coaching!